src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1380,5 +1611,21 @@ private static string BuildErrorMessageForInferredBodyParameter(FactoryContext f
                 .AppendLine();
             return errorMessage.ToString();
         }
+
+        private static string BuildErrorMessageForFormAndJsonBodyParameters(FactoryContext factoryContext)
+        {
+            var errorMessage = new StringBuilder();
+            errorMessage.AppendLine("An action cannot use both form and JSON body parameters.");
+            errorMessage.AppendLine("Below is the list of parameters that we found: ");
+            errorMessage.AppendLine();
+            errorMessage.AppendLine(FormattableString.Invariant($"{"Parameter",-20}| {"Source",-30}"));
+            errorMessage.AppendLine("---------------------------------------------------------------------------------");
+
+            foreach (var kv in factoryContext.TrackedParameters)
+            {
+                errorMessage.AppendLine(FormattableString.Invariant($"{kv.Key,-19} | {kv.Value,-15}"));
+            }
```

Comment:
Nit: Can we share this with logic with BuildErrorMessageForInferredBodyParameter?